File: frontend/src/api/explain.ts
import { isApiConfigured, apiGet } from './client'

export interface SourceCitation {
  source_type: string
  source_id: string
  excerpt: string | null
}

export interface RelationshipFact {
  related_entity_id: string
  related_label: string
  relationship: string
}

export interface ExplainResponse {
  entity_id: string
  case_id: string
  label: string
  narrative: string
  reasoning_steps: string[]
  source_citations: SourceCitation[]
  relationships: RelationshipFact[]
  risk_factors: string[]
}

export async function getEntityExplanation(
  entityId: string,
  caseId: string,
): Promise<ExplainResponse | null> {
  if (!isApiConfigured()) return null
  return apiGet<ExplainResponse>(
    `/explain/${encodeURIComponent(entityId)}?case_id=${encodeURIComponent(caseId)}`,
  )
}

File: backend/app/schemas/explain.py
from pydantic import BaseModel, Field


class SourceCitation(BaseModel):
    source_type: str
    source_id: str
    excerpt: str | None = None


class RelationshipFact(BaseModel):
    related_entity_id: str
    related_label: str
    relationship: str


class ExplainResponse(BaseModel):
    entity_id: str
    case_id: str
    label: str
    narrative: str
    reasoning_steps: list[str] = Field(default_factory=list)
    source_citations: list[SourceCitation] = Field(default_factory=list)
    relationships: list[RelationshipFact] = Field(default_factory=list)
    risk_factors: list[str] = Field(default_factory=list)
